src/Evolution/Systems/CurvedScalarWave/TimeDerivative.hpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Characteristics.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Protocols.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <typename X, typename Symm, typename IndexList>
+class Tensor;
+/// \endcond
+
+namespace CurvedScalarWave {
+/*!
+ * \brief Compute the time derivative of the evolved variables of the
+ * first-order scalar wave system on a curved background.
+ *
+ * The evolution equations for the first-order scalar wave system are given by
+ * \cite Holst2004wt :
+ *
+ * \f{align}
+ * \partial_t\Psi = & (1 + \gamma_1) \beta^k \partial_k \Psi - \alpha \Pi -
+ * \gamma_1 \beta^k \Phi_k \\
+ *
+ * \partial_t\Pi = & - \alpha \gamma^{ij}\partial_i\Phi_j + \beta^k \partial_k
+ * \Pi + \gamma_1 \gamma_2 \beta^k \partial_k \Psi  + \alpha \Gamma^i -
+ * \gamma^{ij} \Phi_i \partial_j \alpha
+ *  + \alpha K \Pi - \gamma_1  \gamma_2 \beta^k \Phi_k\\
+ *
+ * \partial_t\Phi_i = & - \alpha \partial_i \Pi  + \beta^k \partial_k \Phi +
+ * \gamma_2 \alpha \partial_i \Psi - \Pi
+ * \partial_i \alpha + \Phi_k \partial_i \beta^j - \gamma_2 \alpha \Phi_i\\
+ * \f}
+ *
+ * where \f$\Psi\f$ is the scalar field, \f$\Pi\f$ is the
+ * conjugate momentum to \f$\Psi\f$, \f$\Phi_i=\partial_i\Psi\f$ is an
+ * auxiliary variable, \f$\alpha\f$ is the lapse, \f$\beta^k\f$ is the shift,
+ * \f$ \gamma_{ij} \f$ is the spatial metric, \f$ K \f$ is the trace of the
+ * extrinsic curvature, and \f$ \Gamma^i \f$ is the trace of the Christoffel
+ * symbol of the second kind. \f$\gamma_1, \gamma_2\f$ are constraint damping
+ * parameters.
+ */
+template <size_t Dim>
+struct TimeDerivative {
+ public:
+  using temporary_tags = tmpl::list<>;
+
+  template <template <class> class StepPrefix>
+  using return_tags =
```

Comment:
I don't think you need the `return_tags` anymore. These are now known because we don't have to deal with the elliptic solver anymore

Reply:
removed